### DISCUSSION TITLE
How to know when (last) item in ItemSource for an ItemsControl is displayed?

### DISCUSSION THREAD

**Discussion by guldud** [AvaloniaUI/Avalonia#14930] [General]
Are there any best practise to know when a list item comes into view ? I need to load data incrementally when I scroll to the "near end" of the list. 

In Maui theres a https://learn.microsoft.com/da-dk/dotnet/maui/user-interface/controls/collectionview/populate-data?view=net-maui-8.0#load-data-incrementally that does the trick

How would one do something similar in Avalonia?

**Comment by maxkatz6** [comment]
Subscribe on ScrollViewer.ScrollChanged to see when its viewport is close to the end. You can get ScrollViewer from `(ScrollViewer)ListBox.Scroll`.

**Reply by guldud** [reply]
Thanks @maxkatz6
